tests/periph_spi/main.c

```diff
@@ -142,6 +142,30 @@ int cmd_init(int argc, char **argv)
         puts("\t\t2: 1 MHz");
         puts("\t\t3: 5 MHz");
         puts("\t\t4: 10 MHz");
+#ifdef CLOCK_APB1
+        for (int i = 0; i < (int)SPI_NUMOF; i++) {
+            printf("\t\tSPI_DEV(%i), ", i);
+            if (spi_config[i].apbbus == APB1) {
+                clk = CLOCK_APB1;
+                printf("CLOCK_APB1 = %luHz\n", (uint32_t)CLOCK_APB1);
+            }
+            else if (spi_config[i].apbbus == APB2) {
+                clk = CLOCK_APB2;
+                printf("CLOCK_APB2 = %luHz\n", (uint32_t)CLOCK_APB2);
+            }
+            else {
+                printf("error: apbbus configuration ");
+                printf("unrecognized parameter for SPI_DEV(%i)\n", i);
+                return 1;
+            }
+            tmp = 256;
+            for (int j = 0; j < 8; j++) {
+                printf("\t\t\t%i: %i/%i\n", (clk / tmp), clk, tmp);
+                tmp /= 2;
+            }
+            puts("\t\t\t*: other values will be rounded.");
+        }
+#endif
```

Comment:
If you move this to a helper function I might start to like the idea :wink: 
Should be something generic `_print_arch_help()` maybe?

That way it can also be implemented for other platforms and provide use for everyone.

I'm still a bit skeptical if this doesn't lead to duplication of logic between the driver and the help text and that the help text will eventually get out of sync with the implementation in the driver.

Reply:
> I'm still a bit skeptical if this doesn't lead to duplication of logic between the driver and the help text and that the help text will eventually get out of sync with the implementation in the driver.

The help text only show the really obtainable values, and end up with `*: other values will be rounded.` Those values depends on the hardware, not on the implementation in the driver so they cannot be out of sync.